### ISSUE TITLE
[BUG] preview is broken for some code blocks

### ISSUE THREAD

**Issue by lahwaacz** [shd101wyy/vscode-markdown-preview-enhanced#1895]
**Describe the bug**
For some Markdown code blocks, e.g. indented blocks and blocks containing backslash escape codes, the preview does not work correctly - see the screenshot.

**Environment**
- OS: Linux
- VSCode: 1.84.2
- Markdown Preview Enhanced: 0.8.10

**To Reproduce**
Save the Markdown file shown below and open the preview.

**Expected behavior**
Valid Markdown should be parsed and previewed correctly.

**Markdown file**
[markown-preview-bug.md](https://github.com/shd101wyy/vscode-markdown-preview-enhanced/files/13559770/markown-preview-bug.md)

**Screenshots**
![screenshot-2023-12-05@12:16:40](https://github.com/shd101wyy/vscode-markdown-preview-enhanced/assets/1289205/1bda321e-98b5-4921-9f11-647ccad5faba)

**Comment by shd101wyy** [comment]
Yes it's a bug. A temp solution is to write the code block in a new line, like below:

````markdown
- 
  ```python
  def add(a, b):
      return a + b
  ```
````

**Comment by lahwaacz** [comment]
@shd101wyy Thanks, is there also a workaround for the second problem where the code block with `print("\N{COPYRIGHT SIGN}")` is not terminated?

**Comment by KingHowler** [comment]
I also have this exact same bug, although the workaround described above doesn`t work for me

Mine didn`t contain any backslashes or anything like that, one thing I would like you to note is that The code block was using Processing which is a variation of java and C++ and is not a very common language in case that might cause some issues.

![image](https://github.com/user-attachments/assets/c91c883f-6437-4cf4-b8f9-0bbbf179731f)

After testing different things out it seems to me as if your code is not reading that line properly. My solution was to include the "```" twice on seperate lines.

Another interesting thing is that if I open a new markdown file containing only the code block in question it works fine.
Whatever changes I make to the part that came after the "```" didn't affect it.

Here is the file : 
OS: Windows 11 64-bit
VS Code: 1.92.0
Markdown previewer: v 0.8.13

[README.md](https://github.com/user-attachments/files/16530816/README.md)
